engine/core/src/main/java/org/pentaho/reporting/engine/classic/core/layout/build/RichTextStyleResolver.java

```diff
@@ -21,35 +21,40 @@
 import org.pentaho.reporting.engine.classic.core.Section;
 import org.pentaho.reporting.engine.classic.core.layout.style.SimpleStyleSheet;
 import org.pentaho.reporting.engine.classic.core.style.ResolverStyleSheet;
-import org.pentaho.reporting.engine.classic.core.style.resolver.SimpleStyleResolver;
+import org.pentaho.reporting.engine.classic.core.style.css.CSSStyleResolver;
+import org.pentaho.reporting.engine.classic.core.style.resolver.StyleResolver;
 import org.pentaho.reporting.engine.classic.core.util.AbstractStructureVisitor;
+import org.pentaho.reporting.engine.classic.core.function.ProcessingContext;
 
 public class RichTextStyleResolver extends AbstractStructureVisitor {
-  private SimpleStyleResolver simpleStyleResolver;
-  private ResolverStyleSheet resolveStyleSheet;
+  private StyleResolver cssStyleResolver;
+  private static RichTextStyleResolver richResolverInstance;
```

Comment:
This variable is not used anymore, isnt it? However, given that the resolver instance is now tied to the report-process, and given that the report-process is guaranteed to be single-threaded, you can bring back the cached instance of the resolveStyleSheet instead of creating a new one every time. This thing is an aggregator that then gets copied into the SimpleStyleSheet when resolving is complete. Reusing the same instance of it means you dont have to create a new instance (and all its internal arrays) for each element you validate.

Reply:
Yeap, I totally forgot to remove it. I'll update the PR.